src/main/java/org/checkerframework/specimin/PrunerVisitor.java

```diff
@@ -0,0 +1,182 @@
+import com.github.javaparser.ast.body.VariableDeclarator;
+import com.github.javaparser.ast.expr.BooleanLiteralExpr;
+import com.github.javaparser.ast.expr.CharLiteralExpr;
+import com.github.javaparser.ast.expr.DoubleLiteralExpr;
+import com.github.javaparser.ast.expr.Expression;
+import com.github.javaparser.ast.expr.IntegerLiteralExpr;
+import com.github.javaparser.ast.expr.LongLiteralExpr;
+import com.github.javaparser.ast.expr.NullLiteralExpr;
+import com.github.javaparser.ast.type.PrimitiveType;
+import com.github.javaparser.ast.type.Type;
+import com.github.javaparser.ast.visitor.ModifierVisitor;
+import com.github.javaparser.ast.visitor.Visitable;
+import com.github.javaparser.resolution.declarations.ResolvedConstructorDeclaration;
+import com.github.javaparser.resolution.declarations.ResolvedMethodDeclaration;
+import java.util.Iterator;
+import java.util.Set;
+
+/**
+ * This visitor removes every member in the compilation unit that is not a member of its {@link
+ * #methodsToLeaveUnchanged} set or {@link #membersToEmpty} set. It also deletes the bodies of all
+ * methods and replaces them with "throw new Error();" or remove the initializers of fields
+ * (minimized if the field is final) within the {@link #membersToEmpty} set.
+ */
+public class PrunerVisitor extends ModifierVisitor<Void> {
+
+  /**
+   * The methods that should NOT be touched by this pruner. The strings representing the method are
+   * those returned by ResolvedMethodDeclaration#getQualifiedSignature.
+   */
+  private Set<String> methodsToLeaveUnchanged;
+
+  /**
+   * The members, fields and methods, to be pruned. For methods, the bodies are removed. For fields,
+   * the initializers are removed, or minimized in case the field is final. The strings representing
+   * the method are those returned by ResolvedMethodDeclaration#getQualifiedSignature. The strings
+   * representing the field are returned by ResolvedTypeDeclaration#getQualifiedName.
+   */
+  private Set<String> membersToEmpty;
+
+  /**
+   * This is the set of classes used by the target methods. We use this set to determine if we
+   * should keep or delete an import statement. The strings representing the classes are in
+   * the @FullyQualifiedName form.
+   */
+  private Set<String> classesUsedByTargetMethods;
+
+  /**
+   * This boolean tracks whether the element currently being visited is inside a target method. It
+   * is set by {@link #visit(MethodDeclaration, Void)}.
+   */
+  private boolean insideTargetMethod = false;
+
+  /**
+   * Creates the pruner. All methods this pruner encounters other than those in its input sets will
+   * be removed entirely. For both arguments, the Strings should be in the format produced by
+   * ResolvedMethodDeclaration#getQualifiedSignature.
+   *
+   * @param methodsToKeep the set of methods whose bodies should be kept intact (usually the target
+   *     methods for specimin)
+   * @param membersToEmpty the set of members to be empty
```

Comment:
"to be empty" looks like a typo. "to be emptied", maybe? That's the passive voice, though, so it would be better to say "the set of members that this pruner will empty"